src/app.js

```diff
@@ -8,29 +8,119 @@ const port = 3000
 
 app.use(bodyParser.json())
 
-app.post('/', (req, res) => {
+app.post('/spendings', (req, res) => {
   Spending.create({
     createdAt: req.body.createdAt,
-    labels: req.body.labels.split(','),
+    labels: req.body.labels,
     category: req.body.category,
     note: req.body.note,
     amount: req.body.amount,
     currency: req.body.currency,
   }, function (err, spending) {
     if (err) {
-      res.status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR).send({error: 'Internal Error - Internal server error'})
+      res
+        .status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR)
+        .send({error: 'Internal Error - Internal server error'})
     } else {
       console.log(spending)
       res.status(HTTP_STATUS_CODES.OK).send('Successful operation')
     }
   })
 });
 
+const setFilters = (links) => {
+  const filter = {}
+
+  links.forEach( link => {
+    if (link.value) {
+      if(!Object.keys(filter).includes(link.name)) {
+        filter[link.name] = (link.key) ? ( {[link.key]: link.value} ) : link.value
+      }
+      else {
+        const obj = filter[link.name];
+        obj[link.key]=link.value
+      }
+    }
+  })
+
+  return filter
+}
+
+app.get('/spendings', (req, res) => {
+  const {
+    page,
+    perPage,
+    category,
+    labels,
+    start,
+    end
+  } = req.query;
+
+  if (page === undefined) {
+    res
+      .status(HTTP_STATUS_CODES.BAD_REQUEST)
+      .send({ error: '\'page\' is required' });
+    return;
+  }
+
+  if (perPage === undefined) {
+    res
+      .status(HTTP_STATUS_CODES.BAD_REQUEST)
+      .send({ error: '\'perPage\' is required' });
+    return;
+  }
+
+  if (page < 1) {
+    res
+      .status(HTTP_STATUS_CODES.BAD_REQUEST)
+      .send({ error: '\'page\' should be greater than 0' });
+    return;
+  }
+
+  const links = [
+    { name: 'category', value: category },
+    { name: 'labels', value: labels ? labels.split(',') : undefined, key: '$in' },
+    { name: 'createdAt', value: start, key: '$gte' },
+    { name: 'createdAt', value: end, key: '$lte' }
+  ]
+
+  const filter = setFilters(links)
+
+  Spending.countDocuments( filter, (err, count) => {
+    if (err) {
+      res
+        .status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR)
+        .send({error: 'Internal Error - Internal server error'})
+    } else {
+      Spending.find(filter, (err, spendings) => {
+        if (err) {
+          res
+            .status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR)
+            .send({error: 'Internal Error - Internal server error'})
+        } else {
+          res.status(HTTP_STATUS_CODES.OK).send({
+            spendings: spendings,
+            pagination: {
+              spendingCount: count,
```

Comment:
You don't need to send a number of all spendings. Just total pages will be enough